script/test

```diff
@@ -84,22 +89,40 @@ done
 
 shift $(expr $OPTIND - 1)
 
-top_level_filter=$1
+top_level_filter=${1-}
 
 if [[ \
-  -n $TREE_SITTER_TEST_LANGUAGE_FILTER || \
-  -n $TREE_SITTER_TEST_EXAMPLE_FILTER || \
-  -n $TREE_SITTER_TEST_TRIAL_FILTER \
+  -n "$TREE_SITTER_TEST_LANGUAGE_FILTER" || \
+  -n "$TREE_SITTER_TEST_EXAMPLE_FILTER" || \
+  -n "$TREE_SITTER_TEST_TRIAL_FILTER" \
 ]]; then
   : ${top_level_filter:=corpus}
 fi
 
+# Allow to source this script to just set
+# shell environment variables and use `cargo test` directly
+if [ "$0" != "$BASH_SOURCE" ]; then
+    eval set +${-//[is]/}
+    eval set -${shopts//[is]/}
+    cargo() {
+      case "$1" in
+        t|test)
+            command cargo test $test_flags "${@:2}"
+            ;;
+        *)  command cargo "$@"
+            ;;
+      esac
+    }
+    return
+fi
```

Comment:
I don't fully understand this part. You're defining `cargo` as a function, which will then be called in place of the `cargo` executable after sourcing the script?

Could you do without this part? It seems a bit surprising, and specific to your workflow. It seems like you could still source the script just to set all the env vars that are used by the tests.

Reply:
I define this wrapper so it helps to use `cargo test` in any sub folder in the repo and don't think about a path to the `script/test` to call it. There other options like an alias for the git which calls all of it sub commands from the root of the repo, but the wrapper look more comfortable IMO.

And the line `if [ "$0" != "$BASH_SOURCE" ]; then` allows to just source the script into current shell environment without actual execution of it, but set all cargo specifics in the current shell environment.
If you disagree I'll remove this.